### PR TITLE
non-k8s flows blog post

### PR DESCRIPTION
**Pull request by dbentley**

### CODE REVIEW

File: blog/_posts/2020-02-12-local-dev.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+tags:
+  - tilt
+  - local
+keywords:
+  - tilt
+  - local
+---
+
+Many teams that use Kubernetes for production don't use Kubernetes in development to avoid Kubernetes pain points (like the slow inner-loops of building container images). But these teams still feel the pains of multi-service development:
+* Updating the right subset of microservices after you edit files is manual/error-prone/ad hoc even if you're just running bare processes on your laptop.
```
Comment (jazzdan) [inline blog/_posts/2020-02-12-local-dev.md]: It feels weird that this first bullet doesn't have a question in it, but the other two do. I think they should be consistent: either all be questions or statements of problems.

File: blog/_posts/2020-02-12-local-dev.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+* Finding a problem across many services? Playing 20 questions is as clumsy with `tail -f` or `ps` as `kubectl`.
+* Onboarding new users to this clusterf...un? You want them to be productive on day 1 without having to learn the whole stack, regardless of where it's running.
+
+Now you can use Tilt to solve these pains of multi-service development even if you aren't using Kubernetes for local development. Tilt can improve workflows built atop `foreman`, `docker-compose`, `edward`, a custom `start.sh, or a wiki page with tmux instructions.
```
Comment (jazzdan) [inline blog/_posts/2020-02-12-local-dev.md]:
```suggestion
Now you can use Tilt to solve these pains of multi-service development even if you aren't using Kubernetes for local development. Tilt can improve workflows built atop `foreman`, `docker-compose`, `edward`, a custom `start.sh`, or a wiki page with tmux instructions.
```

File: blog/_posts/2020-02-12-local-dev.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+Expanding to support non-Kubernetes development environments gets us closer to our promise: if you use Kubernetes in Prod, you need Tilt for Dev.
```
Comment (jazzdan) [inline blog/_posts/2020-02-12-local-dev.md]: Talking about "promises" seems like a thing we would say internally but not something that makes a lot of sense externally.

Maybe I would rephrase it as: "Now if you use Kubernetes for prod, you should use Tilt for dev. Full stop."

File: blog/_posts/2020-02-12-local-dev.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
+---
+slug: local-dev
+date: 2020-02-11
+author: dan
+layout: blog
+title: "Non-Kubernetes Local Dev"
```
Comment (jazzdan) [inline blog/_posts/2020-02-12-local-dev.md]: Title suggestion:

"Tilt is a Better Foreman"

"Tilt: Supercharged Foreman"

I think we want an attention grabbing title.

File: blog/_posts/2020-02-12-local-dev.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+## Ensuring consistency with `local_resource`
+If you're not using containers for development, you can lose some of the consistency of building an image from a Dockerfile. Adding a local_resource that checks local software installations can highlight incompatibilities before you spend hours debugging a build failure. Write a script called `check_compat.sh` that does the checks and add this to your Tiltfile:
+```python
+local_resource('check_compat', 'check_compat.sh')
+```
+
+Tilt will run this on startup and display an error if the command with an error.
+
+## Try Tilt With Your Project
+Download Tilt and start it. When you add to the Tiltfile, Tilt will automatically pick up the changes. If you want to pair program with a member of the Tilt team to configure Tilt for your local dev, you can [schedule a session](https://calendly.com/dbentley) or check out our [contact info](https://tilt.dev/contact).
```
Comment (jazzdan) [inline blog/_posts/2020-02-12-local-dev.md]: "When you add to the Tiltfile" feels weird. Maybe: "When you change your Tiltfile"?

File: blog/_posts/2020-02-12-local-dev.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  'cd numbers && pip install -r requirements.txt',
+  serve_cmd='cd numbers && python app.py --port=%d' % (numbers_port),
+  # numbers is a flask app that handles most changes via hot reloading, but
+  # not requirements.txt changes
+  deps=['numbers/requirements.txt'])
+```
+
+This example project has clean separation between services. Tilt can handle more complex setups using options to `local_resource` that you can learn about in the [API reference](https://docs.tilt.dev/api.html#api.local_resource).
+
+## Ensuring consistency with `local_resource`
```
Comment (jazzdan) [inline blog/_posts/2020-02-12-local-dev.md]: At first I read this as "Oh, this feature is consistent with `local_resource`, which was a bit confusing.

Maybe reword as: "Check environments with `local_resource`"? "Check pre-reqs with `local_resource`"?

File: blog/_posts/2020-02-12-local-dev.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+## Try Tilt With Your Project
+Download Tilt and start it. When you add to the Tiltfile, Tilt will automatically pick up the changes. If you want to pair program with a member of the Tilt team to configure Tilt for your local dev, you can [schedule a session](https://calendly.com/dbentley) or check out our [contact info](https://tilt.dev/contact).
+
+## Where You Can Go Next
+If you deploy to Kubernetes for production, you have bugs that only show up in Kubernetes. That kind of development, like modifying YAML configuration or fixing networking-specific bugs requires a Kubernetes development workflow. Your team has this workflow as well, but it's probably less understood and much slower.
+
+Fixing those pains is Tilt's sweet spot. Live Update can sync and build into running containers as quickly as running processes locally. Tilt's UI natively displays Kubernetes events and statuses.
+
+Using Tilt for both of these workflows is easy: devs can switch a command-line flag and the same UI that was displaying status of uncontained processes on your laptop is suddenly connected to pods in a dev Kubernetes cluster..
```
Comment (jazzdan) [inline blog/_posts/2020-02-12-local-dev.md]:
```suggestion
Using Tilt for both of these workflows is easy: devs can switch a command-line flag and the same UI that was displaying status of uncontained processes on your laptop is suddenly connected to pods in a dev Kubernetes cluster.
```

Comment (jazzdan) [inline blog/_posts/2020-02-12-local-dev.md]: Maybe link to the config flags docs here?

File: blog/_posts/2020-02-12-local-dev.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+local_resource('check_compat', 'check_compat.sh')
+```
+
+Tilt will run this on startup and display an error if the command with an error.
+
+## Try Tilt With Your Project
+Download Tilt and start it. When you add to the Tiltfile, Tilt will automatically pick up the changes. If you want to pair program with a member of the Tilt team to configure Tilt for your local dev, you can [schedule a session](https://calendly.com/dbentley) or check out our [contact info](https://tilt.dev/contact).
+
+## Where You Can Go Next
+If you deploy to Kubernetes for production, you have bugs that only show up in Kubernetes. That kind of development, like modifying YAML configuration or fixing networking-specific bugs requires a Kubernetes development workflow. Your team has this workflow as well, but it's probably less understood and much slower.
```
Comment (jazzdan) [inline blog/_posts/2020-02-12-local-dev.md]: Maybe mention in here how Tilt helps to alleviate a lot of the painpoints that people have when developing in Kubernetes? The points that were mentioned in the first paragraph?

Comment (dbentley) [reply blog/_posts/2020-02-12-local-dev.md]: The next paragraph is how Tilt has k8s-specific benefits. The first paragraph of the post is about general MDX challenges. I don't think we need to include them here when that's the whole post. WDYT?

File: blog/_posts/2020-02-12-local-dev.md (hunk 4 above)
Comment (dbentley) [reply blog/_posts/2020-02-12-local-dev.md]: WDYT of "Tilt Beyond Kubernetes"?

Argh, I think that's too grandiose...

Comment (dbentley) [reply blog/_posts/2020-02-12-local-dev.md]: But, like, the point is that we're expanding...

File: blog/_posts/2020-02-12-local-dev.md (hunk 8 above)
Comment (jazzdan) [reply blog/_posts/2020-02-12-local-dev.md]: SGTM

File: blog/_posts/2020-02-12-local-dev.md (hunk 4 above)
Comment (jazzdan) [reply blog/_posts/2020-02-12-local-dev.md]: IMO The title's job is to grab your attention to you click in to the article, not to describe the main point of the article. :P

I think "Tilt Beyond Kubernetes" is an improvement, FWIW.
